File: docs/code-examples/box2d_simple.py
"""Log a simple 2D Box."""
import rerun as rr

rr.init("rerun_example_box2d", spawn=True)

rr.log("simple", rr.Boxes2D(mins=[-1, -1], sizes=[2, 2]))

# Log an extra rect to set the view bounds
rr.log("bounds", rr.Boxes2D(sizes=[4.0, 3.0]))

File: docs/code_examples/box2d_simple.rs
//! Log some very simple 2D boxes.

use rerun::{archetypes::Boxes2D, RecordingStreamBuilder};

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let (rec, storage) = RecordingStreamBuilder::new("rerun_example_box2d").memory()?;

    rec.log(
        "simple",
        &Boxes2D::from_mins_and_sizes([(-1., -1.)], [(2., 2.)]),
    )?;

    // Log an extra rect to set the view bounds
    rec.log("bounds", &Boxes2D::from_sizes([(4., 3.)]))?;

    rerun::native_viewer::show(storage.take())?;
    Ok(())
}
